src/transformers/models/mplug_owl/processing_mplug_owl.py

```diff
@@ -0,0 +1,150 @@
+        """
+        return self.tokenizer.decode(*args, skip_special_tokens=skip_special_tokens, **kwargs)
+
+    def tokenize_prompts(self, prompts=None, tokens_to_generate=None, add_BOS=None, rank=0, **kwargs):
+        """Tokenize prompts and make them avaiable on all ranks."""
+
+        # On all ranks set to None so we can pass them to functions
+        prompts_tokens_cuda_long_tensor = None
+        prompts_length_cuda_long_tensor = None
+
+        # On the specified rank, build the above.
+        attention_mask = None
+
+        (
+            prompts_tokens_cuda_long_tensor,
+            prompts_length_cuda_long_tensor,
+            attention_mask,
+        ) = self._tokenize_prompts_and_batch(prompts, tokens_to_generate, add_BOS, **kwargs)
+
+        return {
+            "input_ids": prompts_tokens_cuda_long_tensor,
+            "attention_mask": attention_mask,
+        }
+
+    def _tokenize_prompts_and_batch(self, prompts, tokens_to_generate, add_BOS, **kwargs):
+        import torch
+
+        """Given a set of prompts and number of tokens to generate:
+        - tokenize prompts
+        - set the sequence length to be the max of length of prompts plus the number of tokens we would like to
+          generate
+        - pad all the sequences to this length so we can convert them into a 2D tensor.
+        """
+
+        prompts_tokens = [self._tokenize_prompt(prompt, add_BOS, **kwargs) for prompt in prompts]
+
+        # Now we have a list of list of tokens which each list has a different
+        # size. We want to extend this list to:
+        #   - incorporate the tokens that need to be generated
+        #   - make all the sequences equal length.
+        # Get the prompts length.
+        prompts_length = [len(prompt_tokens) for prompt_tokens in prompts_tokens]
+        # Get the max prompts length.
+        max_prompt_len = max(prompts_length)
+        # Number of tokens in the each sample of the batch.
+        samples_length = max_prompt_len + tokens_to_generate
+        # Now update the list of list to be of the same size: samples_length.
+        for prompt_tokens, prompt_length in zip(prompts_tokens, prompts_length):
+            padding_size = samples_length - prompt_length
+            prompt_tokens.extend([self.tokenizer.eos_token_id] * padding_size)
+
+        # Now we are in a structured format, we can convert to tensors.
+        prompts_tokens_tensor = torch.LongTensor(prompts_tokens)
+        prompts_length_tensor = torch.LongTensor(prompts_length)
+        attention_mask = torch.zeros(prompts_tokens_tensor.shape[:2])
+        for i, l in enumerate(prompts_length_tensor):
+            attention_mask[i, :l] = 1
+        return prompts_tokens_tensor, prompts_length_tensor, attention_mask
+
+    def _tokenize_prompt(self, prompt, add_BOS=False, media_info={"<image>": 65}, **kwargs):
+        media_tokens = {k: -int(i + 1) for i, k in enumerate(media_info.keys())}
```

Comment:
I think the purpose of this method is to properly encode the input text that have the media info tokens?
Can you add some comments on why use this method instead of a simple call to the tokenizer?